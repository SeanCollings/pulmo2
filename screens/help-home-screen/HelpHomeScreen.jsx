import React from 'react';
import { StyleSheet, Text, ScrollView } from 'react-native';

import { COLORS } from '../../constants/constants';
import options from './options';

export const helpHomeScreenOptions = options;

const HelpHomeScreen = (props) => {
  return (
    <ScrollView
      style={styles.container}
      contentContainerStyle={{ alignItems: 'center' }}
    >
      <Text style={styles.textHeading}>Home Page</Text>
      <Text style={styles.textContent}>
        Following the prompt 'click to begin', click on the play button to start
        selected excercise.
      </Text>
      <Text style={styles.textContent}>
        The selected level on the Luft&trade; is displayed below the play button
        for reference and can be updated in your Profile Page.
      </Text>
      <Text style={{ ...styles.textContent, ...styles.textBold }}>
        Please confirm selected level before each excercise.
      </Text>
      <Text style={styles.textContent}>
        The name and summary of the selected excercise is displayed on the page.
        The excercise can be changed by navigating to the 'Excercises' tab at
        the bottom of the screen.
      </Text>
      <Text style={styles.textContent}>
        After completing an excercise, a summary of your workout will be
        displayed. A saved copy of this can be found in your Profile Page.
      </Text>
      <Text style={styles.textContent}></Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 30,
    backgroundColor: COLORS.BACKGROUND,
  },
  textHeading: {
    paddingBottom: 20,
    fontSize: 20,
    color: COLORS.SECONDARY,
    fontFamily: 'tit-regular',
  },
  textContent: {
    fontSize: 16,
    width: '80%',
    textAlign: 'center',
    paddingBottom: 20,
    color: COLORS.TEXT,
    fontFamily: 'tit-light',
  },
  textBold: {
    fontFamily: 'tit-bold',
  },
});

export default HelpHomeScreen;
