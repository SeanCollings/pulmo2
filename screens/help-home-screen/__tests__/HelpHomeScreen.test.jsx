import React from 'react';
import { act, create } from 'react-test-renderer';
import HelpHomeScreen, { helpHomeScreenOptions } from '../HelpHomeScreen';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('HelpHomeScreen - unit test', () => {
  test('should render', async () => {
    await act(async () => {
      tree = create(<HelpHomeScreen />);
    });

    expect(tree).toMatchSnapshot();
  });

  describe('helpHomeScreenOptions', () => {
    test('options', () => {
      const options = helpHomeScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = helpHomeScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });
});
