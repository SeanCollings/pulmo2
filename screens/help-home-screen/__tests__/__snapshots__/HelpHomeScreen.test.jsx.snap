// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpHomeScreen - unit test helpHomeScreenOptions headerTitle 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 20,
    }
  }
>
  Help
</Text>
`;

exports[`HelpHomeScreen - unit test helpHomeScreenOptions options 1`] = `
Object {
  "headerTitle": [Function],
}
`;

exports[`HelpHomeScreen - unit test should render 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 30,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#002f56",
          "fontFamily": "tit-regular",
          "fontSize": 20,
          "paddingBottom": 20,
        }
      }
    >
      Home Page
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Following the prompt 'tap to begin', click on the play button to start the selected excercise.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      The selected level on the Luftâ„¢ is displayed below the play button for reference and can be updated in your Profile Page.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-bold",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Please confirm selected level before each excercise.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      The name and summary of the selected excercise is displayed on the page. The excercise can be changed by navigating to the 'Excercises' tab at the bottom of the screen.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      After starting an excercise, a quick tap of the bottom right button will increment a counter to help you keep track of your inhalation and expiration cycles. A long press will decrement this counter.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      After completing an excercise, a summary of your workout will be displayed. You may review your activity again in your Profile Page.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    />
  </View>
</RCTScrollView>
`;
